examples/lock-app/efr32/include/LockManager.h

```diff
@@ -61,7 +72,9 @@ class LockManager
     } State;
 
     CHIP_ERROR Init(chip::app::DataModel::Nullable<chip::app::Clusters::DoorLock::DlLockState> state,
-                    uint8_t maxNumberOfCredentialsPerUser, uint16_t numberOfSupportedUsers);
+                    uint8_t maxNumberOfCredentialsPerUser, uint16_t numberOfSupportedUsers,
```

Comment:
This is an init function that now takes 5 numeric values (out of which 4 are uint8) as arguments. This is not maintainable. For example:

```cpp
mManager.Init(state, 3, 4, 5, 6, 7);
```

One cannot quickly understand what various values mean.
using constants makes it more obvious however:

```cpp
mManager.Init(state, MAX_CREDENTIALS_PER_USER, MAX_USERS, 
    MAX_YEARDAY_SCHEDULES, 
    MAX_WEEKDAY_SCHEDULES, MAX_HOLIDAY_SCHEDULES);
```

would a reviewer notice that I flipped the order of yearday and weedkay schedules in the init call? I would claim not.

Please create a init params structure with a builder pattern here, so data is clearly identifiable. I would like to read and review code like:

```cpp
mManager.Init(
    state, 
    LockInitParams().SetMaxCredetialsPerUser(10)
                    .SetMaxUsers(20)
                    .SetMaxYeardaySchedules(11)
                    .SetMaxWeedDaySchedules(123)
                    .SetMaxHolidaySchedules(11));
```

or maybe because we have constants, one could do:

```cpp
mManager.Init(state, LockInitParams::DefaultMaxValues())
```

Reply:
Implemented param builder